internal/worker/attempt_lifecycle.go

```
@@ -165,15 +172,21 @@ func (manager *Manager) runAttempt(parent context.Context, claim protocol.Claim,
 	lastResult := ""
 	var finalMessage supervisorMessage
+	attemptStarted := false
 	for index, stage := range stages {
+		if stage.State == protocol.StageSucceeded {
+			lastResult = stage.Result
+			continue
+		}
 		if reason := handle.stopReasonAt(time.Now()); reason != "" {
 			sender.closeAndWait(5 * time.Second)
-			err := stoppedAttemptError(handle, errors.New("Attempt stopped before the next Pipeline stage."))
+			err := stoppedAttemptError(handle, errors.New("Attempt stopped before the next Pipeline stage"))
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)